src/_guides/openapi/code-first.md

```diff
@@ -0,0 +1,177 @@
+
+- [Annotations](#annotations)
+- [OpenAPI-aware Frameworks](#openapi-aware-frameworks)
+- [Traffic Sniffing](#traffic-sniffing)
+- [Move to API Design-first](#move-to-api-design-first)
+
+## Annotations 
+
+The classic approach to the API Code-first workflow is to use code comments (or some other form of annotations) as extensions or plugins to write OpenAPI mixed in with the code it's describing. 
+
+Here's how these annotations look in Go:
+
+```go
+// @title           Swagger Example API
+// @version         1.0
+// @description     This is a sample server celler server.
+// @termsOfService  http://swagger.io/terms/
+
+// @contact.name   API Support
+// @contact.url    http://www.swagger.io/support
+// @contact.email  support@swagger.io
+
+// @host      localhost:8080
+// @BasePath  /api/v1
+
+// @securityDefinitions.basic  BasicAuth
+
+func main() {
+	r := gin.Default()
+
+	c := controller.NewController()
+
+	v1 := r.Group("/api/v1")
+	{
+		accounts := v1.Group("/accounts")
+		{
+			accounts.GET(":id", c.ShowAccount)
+			accounts.GET("", c.ListAccounts)
+			accounts.POST("", c.AddAccount)
+		}
+    //...
+	}
+}
+```
+
+Then the schema level descriptions are mixed in with the code responsible for outputting resources like this:
+
+```go
+type Account struct {
+    ID   int    `json:"id" example:"1"`
+    Name string `json:"name" example:"account name"`
+}
+```
+
+Once the API, endpoints, and resources have all the appropriate annotations there is usually some sort of command you can run to get OpenAPI out of it, and that machine-readable OpenAPI can be used to deploy documentation to Bump.sh or wherever your API documentation lives.
+
+```
+swag init --outputTypes yaml
+
+bump deploy swagger.yaml \
+  --doc my-documentation-name \
+  --token my-documentation-token
+```
+
+This approach has been popular for years, with the main selling point being the idea that keeping OpenAPI metadata near the code will hopefully mean developers keep it up to date update as they work on the code. This is not always the case, which is one of a few reasons this practice is dying out. 
+
+The other is that many of the annotation tools are stuck on older less useful versions of OpenAPI, namely v2.0 instead of v3.0, or the latest and greatest: v3.1.
+
+- **Go** 
+	- [swag](https://github.com/swaggo/swag) (OAS 2.0)
+	- [go-swagger](https://github.com/go-swagger/go-swagger) (OAS 2.0)
+- **PHP**
+	- [Swagger PHP](https://zircote.github.io/swagger-php/) (OAS 3.1 & 3.0)
+- **‌Java/Scala** 
+	- [Swagger Core](https://github.com/swagger-api/swagger-core) (OAS 3.1 & 3.0)  
+- **C#**
+	- [NSwag](https://github.com/RicoSuter/NSwag) (OAS 3.0 & 2.0)
+	- [Swashbuckle](https://github.com/domaindrivendev/Swashbuckle.WebApi) (OAS 2.0)
+- **Node.JS**
```

Comment:
Would you think useful to mention the [`express-openapi`](https://www.npmjs.com/package/express-openapi) lib which we've [already talked about on our blog](https://bump.sh/blog/express-api-openapi)?